src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -76,6 +76,30 @@ public ProgrammingExerciseParticipationResource(ProgrammingExerciseParticipation
         return ResponseEntity.ok(participation.get());
     }
 
+    /**
+     * Get the given student participation with its results and feedbacks.
+     *
+     * @param participationId for which to retrieve the student participation with results and feedbacks.
+     * @return the ResponseEntity with status 200 (OK) and the participation with its results in the body.
+     */
+    @GetMapping("/programming-exercise-participations/{participationId}/student-participation-with-results-and-feedbacks")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
```

Comment:
why should a user be allowed to call this resource?

Reply:
```suggestion
    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
```